prov/rxm/src/rxm_ep.c

```diff
@@ -1350,24 +1324,18 @@ rxm_ep_send_common(struct rxm_ep *rxm_ep, const struct iovec *iov, void **desc,
 
 	assert(count <= rxm_ep->rxm_info->tx_attr->iov_limit);
 
-	ret = rxm_acquire_conn_connect(rxm_ep, dest_addr, &rxm_conn);
+	ret = rxm_ep_send_prerequisites(rxm_ep, dest_addr, &rxm_conn);
 	if (OFI_UNLIKELY(ret))
 		return ret;
 
-	if (OFI_UNLIKELY(!dlist_empty(&rxm_conn->deferred_tx_queue))) {
-		rxm_ep_progress_multi(&rxm_ep->util_ep);
-		if (!dlist_empty(&rxm_conn->deferred_tx_queue))
-			return -FI_EAGAIN;
-	}
-
-	if (data_len <= rxm_ep->rxm_info->tx_attr->inject_size) {
+	if (data_len <= rxm_ep_get_inject_buf_limit(rxm_ep)) {
+		return rxm_ep_inject_send_common(rxm_ep, iov, count, rxm_conn,
+						 context, data, flags, tag, op,
+						 data_len, inject_pkt);
+	} else if (data_len <= rxm_ep_get_eager_buf_limit(rxm_ep)) {
 		struct rxm_tx_eager_buf *tx_buf;
-		size_t total_len = sizeof(struct rxm_pkt) + data_len;
 
-		if (total_len <= rxm_ep->msg_info->tx_attr->inject_size)
-			return rxm_ep_send_inject(rxm_ep, iov, count, rxm_conn,
-						  context, data, flags, tag, op,
-						  data_len, total_len, inject_pkt);
+		assert(!(flags & FI_INJECT));
```

Comment:
This assert appears a few places in this function.  I wonder if this is the best function for this, or if it should be moved down the call stack -- e.g. into normal_send, sar_tx_send, etc.